bundles/org.openhab.extensionservice.marketplace/src/main/java/org/openhab/extensionservice/marketplace/internal/MarketplaceXMLReader.java

```diff
@@ -25,18 +26,19 @@
  * @author Kai Kreuzer - Initial contribution and API
  *
  */
+@NonNullByDefault
 public class MarketplaceXMLReader extends XmlDocumentReader<Marketplace> {
 
     public MarketplaceXMLReader() {
         super.setClassLoader(Marketplace.class.getClassLoader());
     }
 
     @Override
-    public void registerConverters(XStream xstream) {
+    public void registerConverters(@NonNullByDefault({}) XStream xstream) {
     }
 
     @Override
-    public void registerAliases(XStream xstream) {
+    public void registerAliases(@NonNullByDefault({}) XStream xstream) {
```

Comment:
Using `@NonNullByDefault({})` is the only way to be compatible with both annotated an non-annotated classes.

Reply:
I have no problems with this, sometimes it is the only way to deal with libraries.